src/components/UserSelector.tsx

```diff
@@ -1,40 +1,113 @@
+    selectedUser,
+    setSelectedUser,
+    setDetail,
+  } = useContext(UserListContext);
+
+  const dropdownRef = useRef<HTMLDivElement>(null);
+
+  const handleSelectUser = (userId: number) => {
+    setIsLoader(true);
+    setPost([]);
+    if (userId !== undefined) {
+      getPosts(userId)
+        .then(posts => {
+          setPost(posts);
+        })
+        .catch(() => {
+          setErrorPosts(true);
+        })
+        .finally(() => {
+          setIsLoader(false);
+        });
+      setDetail(false);
+
+      setIsDrobdown(false);
+      const usersName = users.find(u => u.id === userId);
+
+      setSelectedUser(usersName || null);
+    }
+  };
+
+  const handleClickOutside = (event: MouseEvent) => {
+    if (
+      dropdownRef.current &&
+      !dropdownRef.current.contains(event.target as Node)
+    ) {
+      setIsDrobdown(false);
+    }
+  };
+
+  const handleDropdown = () => {
+    setIsDrobdown(prev => !prev);
+  };
+
+  function fetchUsers() {
+    getUsers()
+      .then(use => {
+        setUsers(use);
+      })
+      .catch(() => {});
+  }
+
+  useEffect(() => {
+    fetchUsers();
+    document.addEventListener('click', handleClickOutside);
+
+    return () => {
+      document.removeEventListener('click', handleClickOutside);
+    };
+  }, []);
+
   return (
-    <div data-cy="UserSelector" className="dropdown is-active">
+    <div
+      onClick={() => handleClickOutside}
+      ref={dropdownRef}
+      data-cy="UserSelector"
+      className={isDropdown ? 'dropdown is-active' : 'dropdown'}
+    >
       <div className="dropdown-trigger">
         <button
           type="button"
           className="button"
           aria-haspopup="true"
           aria-controls="dropdown-menu"
+          onClick={handleDropdown}
         >
-          <span>Choose a user</span>
+          <span>{selectedUser ? selectedUser.name : 'Choose a user'}</span>
 
           <span className="icon is-small">
             <i className="fas fa-angle-down" aria-hidden="true" />
           </span>
         </button>
       </div>
-
       <div className="dropdown-menu" id="dropdown-menu" role="menu">
         <div className="dropdown-content">
-          <a href="#user-1" className="dropdown-item">
-            Leanne Graham
-          </a>
-          <a href="#user-2" className="dropdown-item is-active">
-            Ervin Howell
-          </a>
-          <a href="#user-3" className="dropdown-item">
-            Clementine Bauch
-          </a>
-          <a href="#user-4" className="dropdown-item">
-            Patricia Lebsack
-          </a>
-          <a href="#user-5" className="dropdown-item">
-            Chelsey Dietrich
-          </a>
+          {users.map(use => (
+            <a
+              href={`#${use.id}`}
+              className={
+                selectedUser && selectedUser.id === use.id
+                  ? 'dropdown-item is-active'
+                  : 'dropdown-item'
+              }
+              key={use.id}
```

Comment:
```suggestion
          {users.map(user => (
            <a
              href={`#${user.id}`}
              className={
                selectedUser && selectedUser.id === user.id
                  ? 'dropdown-item is-active'
                  : 'dropdown-item'
              }
              key={user.id}
```

Reply:
use - to put something such as a tool, skill, or building to a particular purpose